src/components/Header/Header.tsx

```diff
@@ -0,0 +1,121 @@
+import React, { useState } from "react";
+import { UserType } from "@/utils/types/userTypes";
+import { signOut } from "next-auth/react";
+
+interface Props {
+  user?: UserType;
+}
+
+type TabLinks = {
+  [tabName: string]: string;
+};
+
+const Header = (props: Props) => {
```

Comment:
Can we make it so that the prop passed in is user.label which is an enum ["educator", "student", "parent", "administrator"] (or undefined if public)? And then Header deals with the logic of deciding which category of UserType user.label belongs to.

And then in `/`, instead of displaying 3 Headers, just display one that should change when the user updates their role in profile (since this should change user.label)